bluemira/base/parameter_frame/_frame.py

```diff
@@ -164,6 +163,151 @@ def __str__(self) -> str:
+    """Reconstruct unit from its dimensionality"""
+    dim_list = list(map(base_unit_defaults.get, dimensionality.keys()))
+    dim_pow = list(dimensionality.values())
+    return pint.Unit(".".join([f"{j[0]}^{j[1]}" for j in zip(dim_list, dim_pow)]))
+
+
+def _fix_combined_units(unit: pint.Unit) -> pint.Unit:
+    """Converts base unit to a composite unit if they exist in the defaults"""
+    dim_keys = list(combined_unit_dimensions.keys())
+    dim_val = list(combined_unit_dimensions.values())
+    with suppress(ValueError):
+        return pint.Unit(
+            combined_unit_defaults[dim_keys[dim_val.index(unit.dimensionality)]]
+        )
+    return pint.Unit(unit)
+
+
+def _convert_angle_units(
+    modified_unit: pint.Unit, orig_unit_str: str, angle_unit: str
+) -> pint.Unit:
+    """
+    Converts angle units to the base unit default for angles.
+
+    Angles are dimensionless therefore dimensionality conversion
+    from pint doesnt work. Conversions between angle units is also not
+    very robust.
+
+    Parameters
+    ----------
+    modified_unit: pint.Unit
+        reconstructed unit without the angle
+    orig_unit_str: str
+        the user supplied unit (without spaces)
+    angle_unit: str
+        the angle unit in `orig_unit`
+
+    Returns
+    -------
+    pint.Unit;
+        the new unit
+
+    """
+    breaking_units = ["steradian", "square_degree"]
+    new_angle_unit = base_unit_defaults["[angle]"]
+    for b in breaking_units:
+        if b in orig_unit_str:
+            raise NotImplementedError(f"{breaking_units} not supported for conversion")
+    if f"{angle_unit}**" in orig_unit_str:
+        raise NotImplementedError("Exponent angles >1, <-1 are not supported")
+    unit_list = orig_unit_str.split("/", 1)
+    exp = "." if angle_unit in unit_list[0] else "/"
+    modified_unit = "".join(str(modified_unit).split(angle_unit))
+    return pint.Unit(f"{modified_unit}{exp}{new_angle_unit}")
+
+
+def _fix_weird_units(modified_unit: pint.Unit, orig_unit: pint.Unit) -> pint.Unit:
+    """
+    Essentially a crude unit parser for when we have no dimensions
+    or non-comutative dimensions
+    """
+    unit_str = f"{orig_unit:C}"
+
+    ang_unit = [ang for ang in ANGLE_units if ang in unit_str]
+    if len(ang_unit) > 1:
+        raise ValueError(f"No...🤯 {orig_unit}")
+    elif len(ang_unit) == 1:
+        ang_unit = ang_unit[0]
+    else:
+        ang_unit = None
+
+    fpy = "full_power_year" in unit_str
+    dpa = "displacements_per_atom" in unit_str
+
+    if not (fpy or dpa) and ang_unit is None:
+        return pint.Unit(modified_unit)
+
+    new_unit = _non_comutative_unit_conversion(
+        dict(modified_unit.dimensionality), unit_str.split("/")[0], dpa, fpy
+    )
+
+    # Deal with angles
+    return _convert_angle_units(new_unit, unit_str, ang_unit) if ang_unit else new_unit
+
+
+def _non_comutative_unit_conversion(dimensionality, numerator, dpa, fpy):
```

Comment:
What's this doing? Just making sure `dpa` and `fpy` are at the start of the unit?

Reply:
`dpa` is stripped because it is dimensionless and `fpy` will be converted to seconds. They both need to be added back in. i will write a docstring to that effect